### PR TITLE
Clear design class, format code, Python3 portability, user defined mu values

### PR DESCRIPTION
**Pull request by ndem0**
- now the user have the possibility to specify manually the mu values and the files where snapshot are stored, using simple text file; also now it's possible to use a config file for define option like namefile_prefix, ...
- pod.py and cvt.py are completely formatted using pep8 style guide
- class Pod and Cvt are redesign, some methods are splittet and the name methods now are more close to the their function
- interpolation and test_interpolation deleted, looked as useless and old file
- adjusted all the test to new class layout
- now code should runs using Python 3.x (and obviously also using Python 2)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -56,4 +56,4 @@ after_success:
 
 branches:
     only:
-        - master
+        - testing
```
Comment (mtezzele) [inline .travis.yml]: I think here you have to revert to master

File: ezyrb/gui.py [hunk 2]
```diff
@@ -9,11 +9,15 @@
...
 """
 
-import Tkinter
-from tkFileDialog import askopenfilename
+try:
+	import tkinter
+	from tkinter.filedialog import askopenfilename
+except:
+	import Tkinter as tkinter
+	from tkFileDialog import askopenfile
```
Comment (mtezzele) [inline ezyrb/gui.py]: to be coherent with line 14 I suggest to add: as askopenfilename

File: ezyrb/pod.py [hunk 3]
```diff
@@ -2,171 +2,313 @@
...
-		coefs = np.zeros([n_basis,n_points])
+        if plot_singular_values:
+            plt.semilogy(
+                np.linspace(0, eigenvalues.shape[0], eigenvalues.shape[0]),
+                eigenvalues / eigenvalues[0])
+            plt.show()
 
-		for i in range(0,n_points):
-			coefs[:,i] = np.dot(np.transpose(self.pod_basis), self.snapshots[:,i]*self.weights)
+        #nBasis = input('Chose the number of basis functions ')
```
Comment (mtezzele) [inline ezyrb/pod.py]: Please erase useless comments

File: tests/test_cvt.py [hunk 4]
```diff
@@ -81,14 +81,15 @@ def test_cvt_attributes_07b(self):
 		expected_rel_error = 36.3507969
 		np.testing.assert_almost_equal(cvt_handler.rel_error, expected_rel_error)
 		
-		
+	'''
```
Comment (mtezzele) [inline tests/test_cvt.py]: Why this commented test?

File: tests/test_cvt.py [hunk 5]
```diff
@@ -150,9 +151,10 @@ def test_cvt_compute_max_error(self):
 		snapshots = np.load('tests/test_datasets/snapshots_test.npy')
 		weights   = np.load('tests/test_datasets/weights_test.npy')
 		cvt_handler = cvt.Cvt(mu_values, snapshots, pod_basis, weights)
-		cvt_handler.add_new_point()
+		max_error = cvt_handler.get_max_error()
+		print(max_error)
```
Comment (mtezzele) [inline tests/test_cvt.py]: Unnecessary print

File: tests/test_datasets/setting.conf [hunk 6]
```diff
@@ -0,0 +1,3 @@
+[Default]
```
Comment (mtezzele) [inline tests/test_datasets/setting.conf]: Please add some comments in order to easily use the setting.conf

File: tests/test_vtkhandler.py [hunk 7]
```diff
@@ -106,7 +106,7 @@ def test_vtk_write_default_output_name(self):
 		vtk_handler.write(output, outfilename, write_bin=True)
 		os.remove(outfilename)
 
-
+	'''
```
Comment (mtezzele) [inline tests/test_vtkhandler.py]: Why this commented test?

File: tests/test_pod.py [hunk 8]
```diff
@@ -314,25 +335,28 @@ def test_pod_write_structures_03(self):
 		os.remove('pod_basis_Pressure.npy')
 		os.remove('coefs_tria_Pressure.npy')'''
 	
-	
+	'''
```
Comment (mtezzele) [inline tests/test_pod.py]: Why this commented test?

**Comment by mtezzele** [comment]
Also, you have to check if the tutorials are working with the new patch.

**Comment by ndem0** [comment]
After looking for a problem in the tests about writing structure on a file, I see these tests was allready commented on the original version of code, so probably they failed also with "working" version.
By the way, i don't understand the reason to use tabs: as you said, standard is spaces. Do tabs improve your development wprkflow in anyway or are you just an "anticonformist"? ;)

Because in my opinion the best choice is put in the repository a config file for python code formatter (eg. yapf) and all contributors, before commits, run the tool on each file. Code become readable, clean and consistent, using pep8 standard.

**Comment by mtezzele** [comment]
If the tests were already commented ok no problem, I will investigate with Filippo.

Please check the documentation of pod.py. There are members not documented.

As I understood there is another option available in the config file: namefile prefix. Please add that with comments in the file so new users can understand immediately how to use it and all the possibilities.

For the tabs over spaces, since all the code is written with tabs please adhere with the convention (see also https://www.python.org/dev/peps/pep-0008/#tabs-or-spaces). Still, there were commits with mixed indentation, please fix it.

Thank you very much for the config file suggestion, I will think about it.

**Comment by mtezzele** [comment]
Check also the coveralls report in order to identify possible missing tests over the new code.

File: ezyrb/gui.py [hunk 9]
```diff
@@ -37,14 +41,14 @@ class Gui(object):
...
+	:cvar tkinter.Label label_error: label where to print the maximum error on the tesselation.
+	:cvar tkinter.Label label_tria: label where to print the triangulation file path.
 	
-	:cvar Tkinter.Label label_basis: label where to print the pod basis file path.
+	:cvar tkinter.Label label_basis: label where to print the pod basis file path.
 	
-	:cvar Tkinter.Label label_parsing_file: label where to print the parsing file path.
-	:cvar Tkinter.Label label_finish_online: label where to print the finish message.
+	:cvar tkinter.Label label_parsing_file: label where to print the parsing file path.
+	:cvar tkinter.Label label_finish_online: label where to print the finish message.
```
Comment (fsalmoir) [inline ezyrb/gui.py]: Just a curiosity: does this change allow you to make the tkinter.Label linked to the tkinter documentation in the PyGeM documentation?

File: ezyrb/gui.py [hunk 10]
```diff
@@ -98,7 +102,7 @@ def _start_ezyrb_offline(self):
 			self.ezyrb_handler = ez.pod.Pod(self.output_name.get(), self.weights_name.get(), self.namefile_prefix.get(), self.file_format.get())
 		else:
 			#self.ezyrb_handler = ez.interpolation.Interp(output_name, namefile_prefix, file_format)
-			self.ezyrb_handler = ez.interp.Interp(self.output_name.get(), self.namefile_prefix.get(), self.file_format.get())
+			self.ezyrb_handler = ez.interpolation.Interp(self.output_name.get(), self.namefile_prefix.get(), self.file_format.get())
```
Comment (fsalmoir) [inline ezyrb/gui.py]: Now interpolation is used. Please, do not delete it (and its tests).

File: ezyrb/__init__.py [hunk 11]
```diff
@@ -1,11 +1,19 @@
 
-__all__ = ['filehandler', 'matlabhandler', 'vtkhandler', 'cvt', 'pod', 'gui', 'interpolation', 'online']
+__all__ = ['filehandler', 'matlabhandler', 'vtkhandler', 'cvt', 'pod', 'gui', 'online']
```
Comment (fsalmoir) [inline ezyrb/__init__.py]: please, restore interpolation. Also for the try and except below, of course.

File: code_formatter.sh [hunk 12]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
```
Comment (mtezzele) [inline code_formatter.sh]: Some comments would be great

**Comment by mtezzele** [comment]
Thanks for the script! See the video ;) https://www.youtube.com/watch?v=SsoOG6ZeyUI

File: ezyrb/cvt.py [hunk 13]
```diff
@@ -129,7 +129,9 @@ def compute_errors(self):
 				loo_error[j] = np.sqrt(np.sum(np.power(error, 2))
 									   ) / self.rel_error
 			else:
+
 				remaining_mu = np.delete(self.mu_values, j, 1)
+				print(remaining_mu, remaining_snaps)
```
Comment (fsalmoir) [inline ezyrb/cvt.py]: Why did you insert a print?

Comment (ndem0) [inline ezyrb/cvt.py]: Because I shouldn't commit after 11 p.m. but I still do it :D

**Comment by mtezzele** [comment]
@ndem0 squash please with only some meaningful commits, otherwise I will squash everything in only one...

**Pull request merged**